src/components/Popups/RemoveAccessPopup/RemoveAccessPopup.tsx

```diff
@@ -0,0 +1,206 @@
+} from 'styles/styledComponents/AppletImage';
+
+import { Modal } from '../Modal';
+import { buttonTextByStep, getHeadCells } from './RemoveAccessPopup.const';
+import { RemoveAccessPopupProps } from './RemoveAccessPopupProps.types';
+
+export const RemoveAccessPopup = ({
+  removeAccessPopupVisible,
+  onClose,
+  user,
+}: RemoveAccessPopupProps) => {
+  const { t } = useTranslation('app');
+  const { firstName, lastName, email } = user;
+  const applets = user.appletIds.map((id: string) => folders.useApplet(id));
+  const [buttonText, setButtonText] = useState('removeAccess');
+  const [step, setStep] = useState<keyof typeof screenByStep>(1);
+  const [selectedApplets, setSelectedApplets] = useState<string[]>([]);
+  const { control, getValues } = useForm({});
+
+  const getAppletName = useCallback(
+    (id: string) => applets.find((applet) => applet.id === id)?.name || '',
+    [applets],
+  );
+
+  const rows = applets?.map((applet: FolderApplet) => ({
+    name: {
+      content: () => (
+        <StyledFlexTopCenter>
+          {applet.image ? (
+            <StyledSmallAppletImg src={applet.image} alt="Applet image" />
+          ) : (
+            <StyledSmallAppletImgPlaceholder />
+          )}
+          <StyledLabelLarge>{applet.name}</StyledLabelLarge>
+        </StyledFlexTopCenter>
+      ),
+      value: applet.name || '',
+    },
+    actions: {
+      content: () => (
+        <CheckboxController control={control} name={applet.id} label={<></>} defaultValue={''} />
+      ),
+      value: applet.id,
+      width: '200',
+    },
+  }));
+
+  const { execute } = useAsync(revokeAppletUserApi);
+
+  const onSubmit = () => {
+    if (step === 1) {
+      const appletsIds = Object.entries(getValues())
+        .filter((entry) => entry[1])
+        .map(([key]) => key);
+      setSelectedApplets(appletsIds);
+    } else if (step === 2) {
+      selectedApplets.forEach(async (appletId: string) => {
+        await execute({ appletId, profileId: user['_id'], deleteResponse: false });
+      });
+    } else if (step === 3) {
+      onClose();
+
+      return;
+    }
+
+    setStep((prevState) => ++prevState as keyof typeof screenByStep);
+  };
+
+  useEffect(() => setButtonText(buttonTextByStep[step]), [step]);
+
+  const listOfSelectedApplets = (
+    <>
+      {selectedApplets.map((id) => (
+        <StyledTitleBoldMedium key={id}>- {getAppletName(id)} </StyledTitleBoldMedium>
+      ))}
+    </>
+  );
+
+  const screenByStep = {
+    1: (
+      <form noValidate>
+        <StyledBodyLarge sx={{ marginBottom: theme.spacing(2.4) }}>
+          <b>
+            {firstName} {lastName} ({email})
+          </b>
+          {t('userHasAccess')}
+        </StyledBodyLarge>
+        <Table columns={getHeadCells(t)} rows={rows} orderBy="name" uiType={UiType.secondary} />
+      </form>
+    ),
+    2:
+      selectedApplets.length === 1 ? (
+        (() => {
+          const appletName = getAppletName(selectedApplets[0]);
+
```

Comment:
pls move steps to components or func. Also you can use array instead object if you have just 1,2,3 keys